### ISSUE TITLE
Tubesync fails to start service tubesync-init after latest upgrade

### ISSUE THREAD

**Issue by csjoh** [meeb/tubesync#312]
Hi,

As the title indicates, both my Tubesync containers have issues starting the actual app after the latest update on unRAID.

Log:
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service tubesync-init: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
2023-01-19 09:25:56,761 [tubesync/INFO] Using database connection: django.db.backends.mysql://tubesync:[hidden]@192.168.2.8:3306/tubesyncvideo
Operations to perform:
Apply all migrations: admin, auth, background_task, contenttypes, sessions, sync
Running migrations:
s6-sudoc: fatal: unable to get exit status from server: Operation timed out
s6-rc: warning: unable to start service tubesync-init: command exited 111
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
Applying sync.0013_fix_elative_media_file...s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service tubesync-init: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
usermod: no changes
2023-01-20 19:09:11,888 [tubesync/INFO] Using database connection: django.db.backends.mysql://tubesync:[hidden]@192.168.2.8:3306/tubesyncvideo
Traceback (most recent call last):
File "/app/tubesync/dbutils.py", line 25, in ensure_connection
self.connect()
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 200, in connect
self.connection = self.get_new_connection(conn_params)
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/base.py", line 234, in get_new_connection
connection = Database.connect(**conn_params)
File "/usr/local/lib/python3.9/dist-packages/MySQLdb/__init__.py", line 123, in Connect
return Connection(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/MySQLdb/connections.py", line 185, in __init__
super().__init__(*args, **kwargs2)
MySQLdb.OperationalError: (1129, "Host '172.17.0.1' is blocked because of many connection errors; unblock with 'mariadb-admin flush-hosts'")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "/app/manage.py", line 18, in <module>
main()
File "/app/manage.py", line 14, in main
execute_from_command_line(sys.argv)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
utility.execute()
File "/usr/local/lib/python3.9/dist-packages/django/core/management/__init__.py", line 413, in execute
self.fetch_command(subcommand).run_from_argv(self.argv)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/base.py", line 354, in run_from_argv
self.execute(*args, **cmd_options)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/base.py", line 398, in execute
output = self.handle(*args, **options)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/base.py", line 89, in wrapped
res = handle_func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/commands/migrate.py", line 75, in handle
self.check(databases=[database])
File "/usr/local/lib/python3.9/dist-packages/django/core/management/base.py", line 419, in check
all_issues = checks.run_checks(
File "/usr/local/lib/python3.9/dist-packages/django/core/checks/registry.py", line 76, in run_checks
new_errors = check(app_configs=app_configs, databases=databases)
File "/usr/local/lib/python3.9/dist-packages/django/core/checks/database.py", line 13, in check_database_backends
issues.extend(conn.validation.check(**kwargs))
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/validation.py", line 9, in check
issues.extend(self._check_sql_mode(**kwargs))
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/validation.py", line 13, in _check_sql_mode
if not (self.connection.sql_mode & {'STRICT_TRANS_TABLES', 'STRICT_ALL_TABLES'}):
File "/usr/local/lib/python3.9/dist-packages/django/utils/functional.py", line 48, in __get__
res = instance.__dict__[self.name] = self.func(instance)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/base.py", line 405, in sql_mode
sql_mode = self.mysql_server_data['sql_mode']
File "/usr/local/lib/python3.9/dist-packages/django/utils/functional.py", line 48, in __get__
res = instance.__dict__[self.name] = self.func(instance)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/base.py", line 366, in mysql_server_data
with self.temporary_connection() as cursor:
File "/usr/lib/python3.9/contextlib.py", line 117, in __enter__
return next(self.gen)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 603, in temporary_connection
with self.cursor() as cursor:
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 259, in cursor
return self._cursor()
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 235, in _cursor
self.ensure_connection()
File "/app/tubesync/dbutils.py", line 25, in ensure_connection
self.connect()
File "/usr/local/lib/python3.9/dist-packages/django/db/utils.py", line 90, in __exit__
raise dj_exc_value.with_traceback(traceback) from exc_value
File "/app/tubesync/dbutils.py", line 25, in ensure_connection
self.connect()
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 200, in connect
self.connection = self.get_new_connection(conn_params)
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/base.py", line 234, in get_new_connection
connection = Database.connect(**conn_params)
File "/usr/local/lib/python3.9/dist-packages/MySQLdb/__init__.py", line 123, in Connect
return Connection(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/MySQLdb/connections.py", line 185, in __init__
super().__init__(*args, **kwargs2)
django.db.utils.OperationalError: (1129, "Host '172.17.0.1' is blocked because of many connection errors; unblock with 'mariadb-admin flush-hosts'")
s6-rc: warning: unable to start service tubesync-init: command exited 1
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service tubesync-init: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
usermod: no changes
2023-01-20 19:16:40,298 [tubesync/INFO] Using database connection: django.db.backends.mysql://tubesync:[hidden]@192.168.2.8:3306/tubesyncvideo
Traceback (most recent call last):
File "/app/tubesync/dbutils.py", line 25, in ensure_connection
self.connect()
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 200, in connect
self.connection = self.get_new_connection(conn_params)
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/base.py", line 234, in get_new_connection
connection = Database.connect(**conn_params)
File "/usr/local/lib/python3.9/dist-packages/MySQLdb/__init__.py", line 123, in Connect
return Connection(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/MySQLdb/connections.py", line 185, in __init__
super().__init__(*args, **kwargs2)
MySQLdb.OperationalError: (1129, "Host '172.17.0.1' is blocked because of many connection errors; unblock with 'mariadb-admin flush-hosts'")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "/app/manage.py", line 18, in <module>
main()
File "/app/manage.py", line 14, in main
execute_from_command_line(sys.argv)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
utility.execute()
File "/usr/local/lib/python3.9/dist-packages/django/core/management/__init__.py", line 413, in execute
self.fetch_command(subcommand).run_from_argv(self.argv)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/base.py", line 354, in run_from_argv
self.execute(*args, **cmd_options)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/base.py", line 398, in execute
output = self.handle(*args, **options)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/base.py", line 89, in wrapped
res = handle_func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/core/management/commands/migrate.py", line 75, in handle
self.check(databases=[database])
File "/usr/local/lib/python3.9/dist-packages/django/core/management/base.py", line 419, in check
all_issues = checks.run_checks(
File "/usr/local/lib/python3.9/dist-packages/django/core/checks/registry.py", line 76, in run_checks
new_errors = check(app_configs=app_configs, databases=databases)
File "/usr/local/lib/python3.9/dist-packages/django/core/checks/database.py", line 13, in check_database_backends
issues.extend(conn.validation.check(**kwargs))
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/validation.py", line 9, in check
issues.extend(self._check_sql_mode(**kwargs))
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/validation.py", line 13, in _check_sql_mode
if not (self.connection.sql_mode & {'STRICT_TRANS_TABLES', 'STRICT_ALL_TABLES'}):
File "/usr/local/lib/python3.9/dist-packages/django/utils/functional.py", line 48, in __get__
res = instance.__dict__[self.name] = self.func(instance)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/base.py", line 405, in sql_mode
sql_mode = self.mysql_server_data['sql_mode']
File "/usr/local/lib/python3.9/dist-packages/django/utils/functional.py", line 48, in __get__
res = instance.__dict__[self.name] = self.func(instance)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/base.py", line 366, in mysql_server_data
with self.temporary_connection() as cursor:
File "/usr/lib/python3.9/contextlib.py", line 117, in __enter__
return next(self.gen)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 603, in temporary_connection
with self.cursor() as cursor:
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 259, in cursor
return self._cursor()
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 235, in _cursor
self.ensure_connection()
File "/app/tubesync/dbutils.py", line 25, in ensure_connection
self.connect()
File "/usr/local/lib/python3.9/dist-packages/django/db/utils.py", line 90, in __exit__
raise dj_exc_value.with_traceback(traceback) from exc_value
File "/app/tubesync/dbutils.py", line 25, in ensure_connection
self.connect()
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/base/base.py", line 200, in connect
self.connection = self.get_new_connection(conn_params)
File "/usr/local/lib/python3.9/dist-packages/django/utils/asyncio.py", line 33, in inner
return func(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/django/db/backends/mysql/base.py", line 234, in get_new_connection
connection = Database.connect(**conn_params)
File "/usr/local/lib/python3.9/dist-packages/MySQLdb/__init__.py", line 123, in Connect
return Connection(*args, **kwargs)
File "/usr/local/lib/python3.9/dist-packages/MySQLdb/connections.py", line 185, in __init__
super().__init__(*args, **kwargs2)
django.db.utils.OperationalError: (1129, "Host '172.17.0.1' is blocked because of many connection errors; unblock with 'mariadb-admin flush-hosts'")
s6-rc: warning: unable to start service tubesync-init: command exited 1
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service tubesync-init: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
usermod: no changes
2023-01-20 19:18:19,847 [tubesync/INFO] Using database connection: django.db.backends.mysql://tubesync:[hidden]@192.168.2.8:3306/tubesyncvideo
Operations to perform:
Apply all migrations: admin, auth, background_task, contenttypes, sessions, sync
Running migrations:
s6-sudoc: fatal: unable to get exit status from server: Operation timed out
s6-rc: warning: unable to start service tubesync-init: command exited 111
```

I logged into my mariadb container and ran mariadb-admin flush-hosts when I saw the entry in the log, then restarted the container. The part after "unblock with mariadb-admin" is after the restart.

The containers run and I can access the shell, but the app/webui doesn't show.

Help!

**Comment by meeb** [comment]
`django.db.utils.OperationalError: (1129, "Host '172.17.0.1' is blocked because of many connection errors; unblock with 'mariadb-admin flush-hosts'")` is the error so doesn't look like anything to do with tubesync. Your MariaDB server is rejecting connections.

Make sure you're running the `:latest` tubesync, then I guess stop tubesync and fix your MariaDB server with the suggested command, then restart tubesync.

**Comment by csjoh** [comment]
> `django.db.utils.OperationalError: (1129, "Host '172.17.0.1' is blocked because of many connection errors; unblock with 'mariadb-admin flush-hosts'")` is the error so doesn't look like anything to do with tubesync. Your MariaDB server is rejecting connections.
> 
> Make sure you're running the `:latest` tubesync, then I guess stop tubesync and fix your MariaDB server with the suggested command, then restart tubesync.

I've already done that, as I wrote at the bottom of my first post, but tubesync doesn't work even after that.

I've also restarted the mariadb container, then tubesync, but still nothing.

**Comment by meeb** [comment]
Honestly I've no idea here. I don't really use MySQL and never used MariaDB for anything other than testing. The issue is your MariaDB server is rejecting the connections for some reason so the solution is going to be something to do with tweaking MariaDB. I'd stop the tubesync container for now, then try and fix whatever is wrong with MariaDB.

**Comment by csjoh** [comment]
Hello again,

I got it working again, but I don't know why this happened or how.

However:
1. 172.17.0.1 is the docker host. I.e., _every single docker container_ that uses the MariaDB server will be blocked, as the host is routing the traffic from whatever docker IP address the connecting client has, to the MariaDB server.
2. This is a known problem (which I would've found last night if I wasn't too tired and too panic'ed to think straight) and there is a proposed fix in the works over in the MariaDB camp.
3. A workaround for this problem is the following:
* Edit the MariaDB config file and set max_connections = 10000. The default is 100. This allows for more simultaneous connections to the server before it drops new ones. If you're running some sort of monitoring system to check that the server is alive, said monitoring will create a lot of unauthenticated connections which will linger on the server for (by default) 5 minutes. If you check for life every 60 seconds, for instance, you will always have 5 lingering connections to the server. Combine that with a busy server and you're dropping new connections left and right.
* Edit the config file and set max_connect_errors=10000 to allow very many connection errors to occur before the server starts treating the connection errors as something that needs to be dealt with.
* Run mariadb-admin flush-hosts to flush the blocked hosts table.
* Restart the MariaDB server after making those edits, then restart the TubeSync container. If you have more than one, like me, start ONE AT THE TIME and monitor the startup logs. In my case it took a while because there were migrations that were ran. YMMV. Wait for the startup sequence to finish, then check that the webui is back, log in and check that everything works. Keep looking at the logs. Repeat for any other TubeSync instances.

**Comment by meeb** [comment]
Thanks for detailing the steps for any others who might encounter this same issue. Given you're using docker with a standard NAT setup and it seems you're running MariaDB on the host I would guess that MariaDB seeing every single container connecting from the same IP tripped some "bad client IP" limit causing your issue.

Are you happy to have this issue closed?

**Comment by csjoh** [comment]
> Thanks for detailing the steps for any others who might encounter this same issue. Given you're using docker with a standard NAT setup and it seems you're running MariaDB on the host I would guess that MariaDB seeing every single container connecting from the same IP tripped some "bad client IP" limit causing your issue.

The MariaDB server is also running as a docker container with the default port exposed on the host, hence the 192.168.2.8:3306 entry. Not that it matters much in the grand scheme of things, but yes, that's likely what the issue is. I'll keep monitoring the state of things for a few days and report back if anything interesting happens. I don't think it will, and it's not a Tubesync problem either.

One final takeaway: If your Tubesync container stops working (or you notice it not working) a late Friday evening and you're tired and have opened a beer - keep drinking the beer and leave Tubesync alone till the next day when you have the mental capacity to actually interpret what you're looking at. :)

> Are you happy to have this issue closed?

Yes, I am. :)

**Comment by meeb** [comment]
Sage wisdom!
